CodeGeneration/Sources/SyntaxSupport/Node.swift

```diff
@@ -114,18 +114,20 @@ public class Node {
       // any two defined children
       childrenWithUnexpected =
         children.enumerated().flatMap { (i, child) -> [Child] in
+          let childName = child.name.withFirstCharacterUppercased
+
           let unexpectedName: String
           let unexpectedDeprecatedName: String?
 
           if i == 0 {
-            unexpectedName = "UnexpectedBefore\(child.name)"
-            unexpectedDeprecatedName = child.deprecatedName.map { "UnexpectedBefore\($0)" }
+            unexpectedName = "UnexpectedBefore\(childName)"
+            unexpectedDeprecatedName = child.deprecatedType.map { "UnexpectedBefore\($0)" }
```

Comment:
Just like we do `child.name.withFirstCharacterUppercased`, we should also do `child.deprecatedName.withFirstCharacterUppercased`.

And I think below are couple usages of `deprecatedName` that need to be uppercased once we lowercase `deprecatedName`.

Reply:
At least one `Child` has a deprecated name that begins with a lowercase letter, which is inconsistent with the majority that starts with uppercase. This has led to the generation of an unexpected node name like `UnexpectedBeforedeprecatedName`. If we want to avoid changing the API, it seems we must adhere to this implementation.

> And I think below are couple usages of `deprecatedName` that need to be uppercased once we lowercase deprecatedName.

Actually, I've not considered changing `deprecatedName` before. Do you think that we should do that?
I guess it's necessary if we are changing `name`. But what would be the best course of action regarding the aforementioned edge case?

Reply:
> At least one `Child` has a deprecated name that begins with a lowercase

Oh, that’s a bug. The deprecated name of `FunctionType.paramters` should be uppercased. Fixing in https://github.com/apple/swift-syntax/pull/2023.

> Actually, I've not considered changing `deprecatedName` before. 

Yes, absolutely, we should also lowercase `deprecatedName`.